### DISCUSSION TITLE
Website generation and publication

### DISCUSSION THREAD

**Discussion by snazy** [apache/polaris#65] [Ideas]
I'm thinking a bit ahead in time, when a lot more content has been added to the web site and we have per released version  docs (reference, openapi, javadoc, et al).

It's relatively easy to start using the "right" tool for website generation, when there's not much content. It's becomes a pretty big problem to switch to the "right" tool later, when there's a lot of content and permalinks need to be retained.

In Nessie, we started with mkdocs + materials, which was nice and easy. Nowadays, the site generation becomes rather slow and complicated, especially since we added per-version docs. I wish, we'd have used something more advanced.

Generally, whatever tool we use, it has to comply to our OSS license.

Some things that could be useful on the project's web site:
* Good "start page"
* Guides with "how to" information
* Docs - for the latest release, for the "nightly build", for older releases
* Community - contributing and interacting in "all the ways", resources (logos et al), meetings (agendas, etc)
* Blog

For the tool:
* Quick build (ideally) + interactive local dev-preview
* Possible to integrate generated content (reference docs created from source, openapi, javadoc)
* "Properly" usable w/ per release version docs
* Compatible OSS license / no licensing conflicts

I'd like to start the discussion around site content and the "right" tool to use.

What are your thoughts?

/cc @annafil @takidau @RussellSpitzer @jbonofre

**Comment by RussellSpitzer** [comment]
I really don't have tool preferences here, all of the various options have their own limitations but what's really important is content so I'd just want to focus on that.

**Comment by jbonofre** [comment]
Usually I like docusaurus, antora, Hugo. Depending of our next steps :)
